src/main/java/org/zalando/nakadi/domain/Storage.java

```diff
@@ -0,0 +1,135 @@
+package org.zalando.nakadi.domain;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+public class Storage {
+    public enum Type {
+        KAFKA(KafkaConfiguration.class, Timeline.KafkaStoragePosition.class);
+        private final Class configClass;
+        private final Class<? extends Timeline.StoragePosition> positionClass;
+
+        Type(final Class configClass, final Class<? extends Timeline.StoragePosition> positionClass) {
+            this.configClass = configClass;
+            this.positionClass = positionClass;
+        }
+    }
+
+    public static class KafkaConfiguration {
+        @JsonProperty("zk_address")
+        private String zkAddress;
+        @JsonProperty("zk_path")
+        private String zkPath;
+
+        public String getZkAddress() {
+            return zkAddress;
+        }
+
+        public void setZkAddress(final String zkAddress) {
+            this.zkAddress = zkAddress;
+        }
+
+        public String getZkPath() {
+            return zkPath;
+        }
+
+        public void setZkPath(final String zkPath) {
+            this.zkPath = zkPath;
+        }
+
+        @Override
+        public boolean equals(final Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (!(o instanceof KafkaConfiguration)) {
+                return false;
+            }
+
+            final KafkaConfiguration that = (KafkaConfiguration) o;
+            return Objects.equals(zkAddress, that.zkAddress) && Objects.equals(zkPath, that.zkPath);
```

Comment:
And I do not want to get NPE in equals or toString method because someone added annotation `@Nullable` or build object incorrectly. For me these methods should work correctly at any point of time.